packages/engine/src/simulation/controller/run.rs

```diff
@@ -79,7 +75,7 @@ pub async fn sim_run<P: SimulationOutputPersistenceRepr>(
     'sim_main: loop {
         // Behaviors expect context.step() to give the current step rather than steps_taken
         let current_step = steps_taken + 1;
```

Comment:
Do we want to add a span for the current step?
We already can find the step just by reading the logs. 

It will affect some of the tracing-ecosystem. If want to use some service to aggregate timings of the `state_sync` span for example, each one would be considered unique and wouldn't automatically aggregate because the step would differ.

Reply:
My immediate thought is that we don't need it

Reply:
> My immediate thought is that we don't need it and can add it if we find use for it later on

I second this, let's keep it as it is for now and add it later if we need/want it.